### PR TITLE
Migrate AwsSqsSource to use genreconciler

### PR DESCRIPTION
**Pull request by n3wscott**
## Proposed Changes

 - Migrate controller of AwsSqsSource to use genreconciler.
 - Use event types from CRD, don't generate them from the reconciler.

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: awssqs/config/logging.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        "messageKey": "msg",
+        "stacktraceKey": "stacktrace",
+        "lineEnding": "",
+        "levelEncoder": "",
+        "timeEncoder": "iso8601",
+        "durationEncoder": "",
+        "callerEncoder": ""
+      }
+    }
+
```
Comment (mattmoor) [inline awssqs/config/logging.yaml]: Remove trailing whitespace:
```suggestion

```

Comment (mattmoor) [inline awssqs/config/logging.yaml]: Remove trailing whitespace:
```suggestion

```

File: awssqs/pkg/apis/sources/v1alpha1/aws_sqs_types.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ type AwsSqsSourceSpec struct {
 	// Sink is a reference to an object that will resolve to a domain name to
 	// use as the sink.  This is where events will be received.
 	// +optional
-	Sink *corev1.ObjectReference `json:"sink,omitempty"`
+	Sink *corev1.ObjectReference `json:"sink,omitempty"` // TODO this is not the source duck anymore
```
Comment (n3wscott) [inline awssqs/pkg/apis/sources/v1alpha1/aws_sqs_types.go]: AqsAqsSource does not conform to the source ducktype.

Comment (n3wscott) [reply awssqs/pkg/apis/sources/v1alpha1/aws_sqs_types.go]: Tracking: https://github.com/knative/eventing-contrib/issues/949

File: awssqs/config/logging.yaml (hunk 1 above)
Comment (mattmoor) [inline awssqs/config/logging.yaml]: Remove trailing whitespace:
```suggestion

```

Comment (mattmoor) [inline awssqs/config/logging.yaml]: Remove trailing whitespace:
```suggestion

```

File: awssqs/pkg/reconciler/awssqssource.go [hunk 3]
```diff
@@ -215,50 +142,13 @@ func (r *reconciler) getReceiveAdapter(ctx context.Context, src *v1alpha1.AwsSqs
...
-func (r *reconciler) removeFinalizer(s *v1alpha1.AwsSqsSource) {
-	finalizers := sets.NewString(s.Finalizers...)
-	finalizers.Delete(finalizerName)
-	s.Finalizers = finalizers.List()
-}
-
 func getLabels(src *v1alpha1.AwsSqsSource) map[string]string {
 	return map[string]string{
-		"knative-eventing-source":      controllerAgentName,
+		"knative-eventing-source":      "awssqssource.sources.eventing.knative.dev",
```
Comment (lberk) [inline awssqs/pkg/reconciler/awssqssource.go]: in other sources we use `eventing.knative.dev/source: controllerAgentName` -- what's the right approach here?
and if it's your implementation here, do we want to use `sources.eventing.knative.dev` ? or just `sources.eventing.dev` ?

File: awssqs/pkg/reconciler/awssqssource.go [hunk 4]
```diff
@@ -18,148 +18,89 @@ package reconciler
...
+		return nil
 	}
-	src.Status.MarkSink(sinkURI)
 
-	_, err = r.createReceiveAdapter(ctx, src, sinkURI)
-	if err != nil {
-		logger.Error("Unable to create the receive adapter", zap.Error(err))
+	if _, err := r.createReceiveAdapter(ctx, src); err != nil {
+		logging.FromContext(ctx).Error("Unable to create the receive adapter", zap.Error(err))
 		return err
```
Comment (lberk) [inline awssqs/pkg/reconciler/awssqssource.go]: In other genreconciler based reconcilers, we seem to also have a `newDeploymentFailed()` function returned which also passes the error, should we also use that here?  Shouldn't we also call `src.Status.MarkNotDeployed()`?

File: awssqs/pkg/reconciler/awssqssource.go (hunk 3 above)
Comment (lberk) [reply awssqs/pkg/reconciler/awssqssource.go]: ah, I'm also realizing that we haven't migrated awssqs to `sources.knative.dev` submitted (will rebase on these changes as needed) #956

Comment (n3wscott) [reply awssqs/pkg/reconciler/awssqssource.go]: Yeah I did not change this in this migration.

File: awssqs/pkg/reconciler/awssqssource.go (hunk 4 above)
Comment (n3wscott) [reply awssqs/pkg/reconciler/awssqssource.go]: This is now what this did originally. There is lots to improve in this controller. That is not my aim in the PR. Just moving what was there to `+genreconciler`

File: awssqs/pkg/reconciler/awssqssource.go (hunk 3 above)
Comment (lberk) [reply awssqs/pkg/reconciler/awssqssource.go]: makes sense, but if we're going to change the approach of `controllerAgentName` to `awssqssource.sources.eventing.knative.dev` should we not keep the overall label map consistent (wondering which approach is right, the apply that one moving forward)

Comment (n3wscott) [reply awssqs/pkg/reconciler/awssqssource.go]: This is the scheme used in genreconciler

**Pull request merged**